### PR TITLE
Highlight target drop control when dragging

### PR DESCRIPTION
**Pull request by dangmarm**

### CODE REVIEW

File: src/components/layout-editor-tool/layout-editor-tool-bar/layout-editor-tool-bar.tsx [hunk 1]
```diff
@@ -55,17 +59,28 @@ export class LayoutEditorToolBar {
...
   }
 
+  private initialize() {
+    const isDragTarget = isControlDragActiveTarget(this.control);
+
+    this.initializeButtonSelect();
+
+    this.dragHidden = isDragTarget || !isControlDraggable(this.control);
+    this.buttonSelectHidden =
+      isDragTarget || (this.parentControl ? false : true);
```
[the remark this answers is not in the document]
Comment (dangmarm) [reply src/components/layout-editor-tool/layout-editor-tool-bar/layout-editor-tool-bar.tsx]: ok

File: src/components/layout-editor-tool/layout-editor-tool-highlight/layout-editor-tool-highlight-controller.tsx [hunk 2]
```diff
@@ -81,19 +103,31 @@ export class LayoutEditorToolHighlightController {
...
         />
         {this.selectedControls.map(control => (
           <gx-le-tool-selection
+            key={`highlight-selected-${control.id}`}
             control={control}
             changeHighlight
             loadBar
             loadBox
             preview={this.preview}
+            hidden={this.dragging ? true : false}
```
[the remark this answers is not in the document]
Comment (dangmarm) [reply src/components/layout-editor-tool/layout-editor-tool-highlight/layout-editor-tool-highlight-controller.tsx]: ok

File: src/components/layout-editor-tool/layout-editor-tool-highlight/layout-editor-tool-highlight-controller.tsx [hunk 3]
```diff
@@ -81,19 +103,31 @@ export class LayoutEditorToolHighlightController {
     return (
       <div>
         <gx-le-tool-selection
+          key="highlight-hovered"
           control={this.hoveredControl}
           changeSmooth
           preview={this.preview}
+          hidden={this.dragging ? true : false}
```
[the remark this answers is not in the document]
Comment (dangmarm) [reply src/components/layout-editor-tool/layout-editor-tool-highlight/layout-editor-tool-highlight-controller.tsx]: ok

File: src/components/layout-editor/layout-editor-control-resolver.tsx [hunk 4]
```diff
@@ -70,12 +70,12 @@ export function getControlWrapperCommonAttrs(
     "data-gx-le-control-name": control["@controlName"] || "",
     "data-gx-le-control-type-name":
       (controlDefinition && controlDefinition.typeName) || "Unknown control",
-    draggable: "true"
+    draggable: control.isRootControl ? "false" : "true"
```
[the remark this answers is not in the document]
Comment (dangmarm) [reply src/components/layout-editor/layout-editor-control-resolver.tsx]: ok

File: src/components/layout-editor/layout-editor-control-resolver.tsx [hunk 5]
```diff
@@ -70,12 +70,12 @@ export function getControlWrapperCommonAttrs(
...
   };
 }
 
-export function getControlCommonAttrs() {
+export function getControlCommonAttrs(control) {
   return {
-    draggable: true
+    draggable: control.isRootControl ? false : true
```
[the remark this answers is not in the document]
Comment (dangmarm) [reply src/components/layout-editor/layout-editor-control-resolver.tsx]: ok

File: src/components/metadata/transform.ts [hunk 6]
```diff
@@ -14,20 +14,24 @@ function transformLayout(
...
 ): GeneXusAbstractLayout.IContainer {
   const transformControlFn = getTransformFunctionByType(childControlType);
 
   return {
     childControlType,
     controlType: childControlType,
-    [childControlType]: transformControlFn(rawContainer[childControlType])
+    [childControlType]: {
+      ...transformControlFn(rawContainer[childControlType]),
+      ...(isRoot && { isRootControl: true })
```
[the remark this answers is not in the document]
Comment (dangmarm) [reply src/components/metadata/transform.ts]: Spread of boolean (...false) is valid when merge objects but not for arrays ([...false]).

File: src/components/layout-editor/layout-editor.tsx [hunk 7]
```diff
@@ -111,6 +115,16 @@ export class LayoutEditor {
...
 
+  /**
+   * Fired when a control is moved over valid target control
+   *
+   * | Property         | Details                                                                                                          |
+   * | ---------------- | ---------------------------------------------------------------------------------------------------------------- |
+   * | `targetControlId`| Identifier of the control where the control was hovered                                                          |
+   *
+   */
+  @Event() moveTarget: EventEmitter;
```
[the remark this answers is not in the document]
Comment (dangmarm) [reply src/components/layout-editor/layout-editor.tsx]: Propose others:

dropTargetChanged
moveTargetChanged
moveDropTargetChanged
moveActiveTargetChanged

[the remark this answers is not in the document]
Comment (ncardeli) [reply src/components/layout-editor/layout-editor.tsx]: I like `dropTargetChanged`

File: src/components/layout-editor-tool/layout-editor-tool-highlight/layout-editor-tool-highlight-controller.tsx [hunk 8]
```diff
@@ -40,6 +48,20 @@ export class LayoutEditorToolHighlightController {
       },
       { passive: true }
     );
+    this.editor.element.addEventListener(
+      "dragstart",
+      () => {
+        this.dragging = true;
+      },
```
Comment (ncardeli) [inline src/components/layout-editor-tool/layout-editor-tool-highlight/layout-editor-tool-highlight-controller.tsx]: When there's only one line in the arrow function, you can use a more compact syntax:

`() => this.dragging = true`

Comment (dangmarm) [reply src/components/layout-editor-tool/layout-editor-tool-highlight/layout-editor-tool-highlight-controller.tsx]: Done

**Pull request merged**